onnxruntime/contrib_ops/webgpu/quantization/subgroup_matrix_matmul_nbits.cc

```diff
@@ -1,18 +1,187 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+#include <tuple>
+
 #include "contrib_ops/webgpu/quantization/subgroup_matrix_matmul_nbits.h"
 
 namespace onnxruntime {
 namespace contrib {
 namespace webgpu {
 
-Status SubgroupMatrixMatMulNBitsProgram::GenerateShaderCode(ShaderHelper& shader) const {
-  shader.AddInput("input_a", ShaderUsage::UseUniform | ShaderUsage::UseIndicesTypeAlias | ShaderUsage::UseValueTypeAlias);
-  shader.AddInput("input_b", ShaderUsage::UseUniform);
-  shader.AddInput("scales_b", ShaderUsage::UseUniform);
-  shader.AddOutput("output", ShaderUsage::UseUniform | ShaderUsage::UseElementTypeAlias);
+// The elements' sequence must match the enum class SubgroupMatrixComponentType in webgpu_cpp.h
+constexpr std::string_view ComponentTypeName[] = {"unknown", "f32", "f16", "u32", "i32"};
+
+// std::tuple<architecture, backendType, componentType, resultComponentType, M, N, K, subgroupMinSize, subgroupMaxSize>
+static const std::tuple<std::string_view, wgpu::BackendType, wgpu::SubgroupMatrixComponentType, wgpu::SubgroupMatrixComponentType,
+                        uint32_t, uint32_t, uint32_t, uint32_t, uint32_t>
+    intel_supported_subgroup_matrix_configs[] = {
+        {"xe-2lpg", wgpu::BackendType::Vulkan, wgpu::SubgroupMatrixComponentType::F16, wgpu::SubgroupMatrixComponentType::F16, 8, 16, 16, 16, 32},
+        {"xe-2lpg", wgpu::BackendType::Vulkan, wgpu::SubgroupMatrixComponentType::F16, wgpu::SubgroupMatrixComponentType::F32, 8, 16, 16, 16, 32}};
 
+bool IsSubgroupMatrixConfigSupportedOnIntel(onnxruntime::webgpu::ComputeContext& context, int32_t& config_index) {
```

Comment:
it looks like this check can be done only once in initialization?

Reply:
It may be just fine for this PR - I am thinking of a refactoring plan for feature detection, including a subgroup matrix helper and dp4a matrix helper. but it's not necessary to be a part of the current PR.